### PR TITLE
Improvements from EWC lokale klimaatinformatie: Colors, CSV, Flexible config

### PR DESCRIPTION
**Pull request by maartenplieger**
## Fixes encountered during EWC lokale klimaatinformatie work

**Version 2.17.0 2024-02-14**

- Colors in GetMap images now matches exactly the colors configured in the legend and style in png32 mode
- CSV reader is more flexible
- Sub configuration files (via include) can now substitute variables as well

### CODE REVIEW

**Review by maartenplieger — COMMENTED** [review]
All OK

**Comment by lukas-phaf** [comment]
Not specific to this branch, but if it is an easy fix...

Screenshot 2023-12-13 at 4 12 55 PM

File: adagucserverEC/CConvertGeoJSON.cpp [hunk 1]
```diff
@@ -93,14 +93,14 @@ void CConvertGeoJSON::drawpolyWithHoles_index(int xMin, int yMin, int xMax, int
   int IMAGE_RIGHT = xMax;
   int scanLineWidth = IMAGE_RIGHT - IMAGE_LEFT;
 
-  int cntLines = 0;
-  int cntHoleLists = 0;
+//  int cntLines = 0;
```
Comment (maartenplieger) [inline adagucserverEC/CConvertGeoJSON.cpp]: Will remove

File: adagucserverEC/CColor.h [hunk 2]
```diff
@@ -47,7 +47,7 @@ class CColor {
   CColor(const char *color) { parse(color); }
   CT::string c_str() {
     CT::string r;
-    r.print("#%s%s%s%s", CT::string::getHex(this->r), CT::string::getHex(this->g), CT::string::getHex(this->b), CT::string::getHex(this->a));
+    r.print("#%s%s%s%s", CT::string::getHex(this->r).c_str(), CT::string::getHex(this->g).c_str(), CT::string::getHex(this->b).c_str(), CT::string::getHex(this->a).c_str());
```
Comment (maartenplieger) [inline adagucserverEC/CColor.h:50]: Check!

File: adagucserverEC/CImgRenderPolylines.cpp [hunk 3]
```diff
@@ -218,7 +218,7 @@ void CImgRenderPolylines::render(CImageWarper *imageWarper, CDataSource *dataSou
           // Draw the polygon holes
           if (true) {
             std::vector<PointArray> holes = itpoly->getHoles();
-            int h = 0;
+//            int h = 0;
```
Comment (belentorrente) [inline adagucserverEC/CImgRenderPolylines.cpp:221]: Will we need this h in the future, or can it be removed?

File: tests/expectedoutputs/TestWMS/test_WMSGetCapabilities_KMDS_PointNetCDF_pointstylepoint.xml [hunk 4]
```diff
@@ -15,7 +15,7 @@
         <KeywordList>
             <Keyword>view</Keyword>
             <Keyword>infoMapAccessService</Keyword>
-            <Keyword>ADAGUCServer version 2.13.7, of Nov  1 2023 11:45:57 </Keyword>
+            <Keyword>ADAGUCServer version 2.14.0, of Nov 30 2023 10:09:29 </Keyword>
```
Comment (belentorrente) [inline tests/expectedoutputs/TestWMS/test_WMSGetCapabilities_KMDS_PointNetCDF_pointstylepoint.xml:18]: Shouldn't this show the latest version of the server?

**Review by belentorrente — APPROVED** [review]

**Pull request merged**